src/jit/morph.cpp

```diff
@@ -7731,12 +7720,16 @@ void Compiler::fgMorphTailCall(GenTreeCall* call)
     if (call->IsVirtualStub())
     {
         flags = CORINFO_TAILCALL_STUB_DISPATCH_ARG;
-
+#ifndef LEGACY_BACKEND
+        GenTree* stubAddrArg = fgGetStubAddrArg(call);
+        // And push the stub address onto the list of arguments
```

Comment:
Couldn't you change `stubAddrArg` to just `arg` and then tail merge the next two lines of this `#if` with the same two lines of the `#endif`?

Reply:
We will delete the legacy soon, so I think it is better to keep this change that way (do not want to move `GenTree* stubAddrArg` out of `ifdef` or do `call->gtCallArgs = gtNewListNode(stubAddrArg, call->gtCallArgs);` that uses `stubAddrArg` that is defined twice (under different `#ifdefs`).